src/Course/List.hs

```diff
@@ -146,17 +131,17 @@ map =
 -- prop> \x -> filter (const True) x == x
 --
 -- prop> \x -> filter (const False) x == Nil
-filter ::
-  (a -> Bool)
-  -> List a
-  -> List a
-filter =
-  error "todo: Course.List#filter"
+filter :: (a -> Bool) -> List a -> List a
+filter f = foldRight reducer Nil
+  where
+    reducer x acc
```

Comment:
nice use of guards :) someone hates ifs ;-)

Reply:
this might be interesting to you to think about:

```
filter f = foldrRight reducer Nil
  where 
      reducer x 
        | f x -> (x :.) 
        | otherwise id
```

Reply:
yeah, that's me the if hater :P 

interesting indeed! I'll test it out later